<!-- Run the aruco_detect node -->
<launch>
    
    <!-- namespace for nuc input -->
    <arg name="nuc_name" default="/nuc1"/>
    <arg name="nuc_address" default="" />
    <arg name="nuc_user" default="" />
    <arg name="nuc_password" default="" />
    <arg name="env_loader_path" default="/opt/ros/noetic/env_remote.sh" />


    <arg name="image" default="image"/>
    <arg name="transport" default="raw"/>
    <arg name="fiducial_len" default="0.035"/>
    <arg name="dictionary" default="0"/>
    <arg name="do_pose_estimation" default="true"/>
    <arg name="vis_msgs" default="false"/>
    <arg name="ignore_fiducials" default="" />
    <arg name="fiducial_len_override" default="" />
    <arg name="verbose" default="false"/>

    <machine
        name="$(arg nuc_name)"
        address="$(arg nuc_address)"
        env-loader="$(arg env_loader_path)"
        user="$(arg nuc_user)"
        password="$(arg nuc_password)"

    />
    <node
        machine="$(arg nuc_name)"
        pkg="aruco_detect"
        name="aruco_detect"
        type="aruco_detect"
        output="screen"
        respawn="false">

        <param name="image_transport" value="$(arg transport)"/>
        <param name="publish_images" value="true" />
        <param name="fiducial_len" value="$(arg fiducial_len)"/>
        <param name="dictionary" value="$(arg dictionary)"/>
        <param name="do_pose_estimation" value="$(arg do_pose_estimation)"/>
        <param name="vis_msgs" value="$(arg vis_msgs)"/>
        <param name="ignore_fiducials" value="$(arg ignore_fiducials)"/>
        <param name="fiducial_len_override" value="$(arg fiducial_len_override)"/>
        <param name="verbose" value="$(arg verbose)"/>
        <remap from="camera" to="$(arg nuc_name)/$(arg image)_$(arg transport)"/>
        <remap from="camera_info" to="$(arg nuc_name)/detect/camera_info"/>
        <!-- <remap from="/fiducial_transforms" to="/$(arg nuc_name)/fiducial_transforms"/>
        <remap from="/fiducial_poses" to="/$(arg nuc_name)/fiducial_poses"/>
        <remap from="/fiducial_images" to="/$(arg nuc_name)/fiducial_images"/>
        <remap from="/fiducial_images/compressed" to="/$(arg nuc_name)/fiducial_images/compressed"/>
        <remap from="/fiducial_images/theora" to="/$(arg nuc_name)/fiducial_images/theora"/>
        <remap from="/fiducial_images/compressedDepth" to="/$(arg nuc_name)/fiducial_images/compressedDepth"/>
        <remap from="/fiducial_images/theoraDepth" to="/$(arg nuc_name)/fiducial_images/theoraDepth"/>
        <remap from="/fiducial_images/depth" to="/$(arg nuc_name)/fiducial_images/depth"/>
        <remap from="/fiducial_images/depth/compressed" to="/$(arg nuc_name)/fiducial_images/depth/compressed"/>
        <remap from="/fiducial_images/depth/theora" to="/$(arg nuc_name)/fiducial_images/depth/theora"/>
        <remap from="/fiducial_images/depth/compressedDepth" to="/$(arg nuc_name)/fiducial_images/depth/compressedDepth"/> -->
  </node>
</launch>